### PR TITLE
Push in front of list instead of appending to end

### LINKED ISSUE #140: Optimize breadcrumb list creation
Currently this is done by appending to the end of the list [source](https://github.com/getsentry/sentry-elixir/blob/9077b52c5b25f257d53f58126d671343471b4142/lib/sentry/context.ex#L64) which is quite inefficient due to list construction. Instead we should append to the beginning and reverse list only when needed. This should provide some performance gain, allow simple removal of breadcrumbs and would make code more idiomatic overall.

### PR DESCRIPTION
**Pull request by hauleth**
Close #140

### CODE REVIEW

File: lib/sentry/context.ex [hunk 1]
```diff
@@ -54,16 +54,15 @@ defmodule Sentry.Context do
   end
 
   defp get_context do
-    Process.get(@process_dictionary_key) || %{}
+    Process.get(@process_dictionary_key, %{})
```
Comment (mitchellhenke) [inline lib/sentry/context.ex]: These are unfortunately not the same, and should stay as is.

Comment (jeregrine) [inline lib/sentry/context.ex]:
```elixir
iex(1)> Process.get(:hello, [])    
[]
iex(2)> Process.get(:hello)    
nil
iex(3)> Process.get(:hello) || []
[]
iex(4)> Process.put(:hello)
** (UndefinedFunctionError) function Process.put/1 is undefined or private. Did you mean one of:

      * put/2

    (elixir) Process.put(:hello)
iex(4)> Process.put(:hello, nil)
nil
iex(5)> Process.get(:hello) || []
[]
iex(6)> Process.get(:hello, [])  
nil
iex(7)> 
```

Comment (hauleth) [inline lib/sentry/context.ex]: Reverted

**Comment by mitchellhenke** [comment]
Thank you! 💖

**Pull request merged**